node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentContextImpl.java

```diff
@@ -97,4 +119,66 @@ public void log(Logger logger, Level level, String message) {
     public void log(Logger logger, Level level, String message, Throwable throwable) {
         logger.log(level, logPrefix + message, throwable);
     }
+    
+
+    /** For testing only! */
+    public static class Builder {
```

Comment:
Does it have to be public?

Reply:
Yes, this is used in som `vespa/hosted` tests as well